### PR TITLE
Add pino.final

### PR DESCRIPTION
**Pull request by alferpal**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: http://getpino.io/#/docs/api?id=pino-final
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

This PR adds the definition for the pino.final functions, as seen here: http://getpino.io/#/docs/api?id=pino-final

This one's been somewhat tricky because of incomplete / not entirely accurate documentation, but by looking at the code I'd say the definition matches what the code does:
http://getpino.io/#/docs/extreme?id=log-loss-prevention
https://github.com/pinojs/pino/blob/master/lib/tools.js#L305

### CODE REVIEW

**Review by screendriver — CHANGES_REQUESTED** [review]
Could you please fix the import as we discussed [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30131#issuecomment-434614196)?

**Comment by alferpal** [comment]
@screendriver yes, will update the PR with that in a few hours or so ( at most a couple of days)

**Comment by alferpal** [comment]
@screendriver The import should be fixed now

File: types/sonic-boom/index.d.ts [hunk 1]
```diff
@@ -7,9 +7,7 @@
 
 import { EventEmitter } from 'events';
 
-export = SonicBoom;
```
Comment (screendriver) [inline types/sonic-boom/index.d.ts]: `SonicBoom` is a CommonJS module (as you can see [here](https://github.com/mcollina/sonic-boom/blob/master/index.js#L219)) and should be export as before `export = SonicBoom`

File: types/sonic-boom/sonic-boom-tests.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import SonicBoom = require('sonic-boom');
```
Comment (screendriver) [inline types/sonic-boom/sonic-boom-tests.ts]: To verify the correct export (see my other comment) you have to leave that line `import SonicBoom = require('sonic-boom');` as it was before.

**Review by screendriver — CHANGES_REQUESTED** [review]

File: types/pino/index.d.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@
 import * as stream from 'stream';
 import * as http from 'http';
 import { EventEmitter } from 'events';
-import SonicBoom = require('sonic-boom');
```
Comment (screendriver) [inline types/pino/index.d.ts]: Sorry for the confusion! I just saw that `sonic-boom` is a CommonJS module (see my comment below) and **your** initial import was correct: `import SonicBoom = require('sonic-boom');` but the other ones above were wrong. Could you change the other imports to `import ... = require()` as well? Many thanks 👍

**Review by screendriver — CHANGES_REQUESTED** [review]

File: types/pino/index.d.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@
...
+import SonicBoom from 'sonic-boom';
```
Comment (screendriver) [inline types/pino/index.d.ts]: Same here. Please do a `import SonicBoom = require('sonic-boom');`

File: types/pino/index.d.ts (hunk 3 above)
Comment (alferpal) [reply types/pino/index.d.ts]: Sure! Should be fixed now

**Review by screendriver — APPROVED** [review]
LGTM 👍

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**